sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -73,6 +74,7 @@ class Analyzer(
       ResolveGroupingAnalytics ::
       ResolvePivot ::
       ResolveUpCast ::
+      ResolveAggregateFunctions ::
```

Comment:
In this rule `ResolveAggregateFunctions`, it has a case for pushing down the expressions in `Sort` operator to the underlying `Aggregate`. This needs to be done earlier than the rule `ResolveSortReferences`; Otherwise, we might add unnecessary missing sorting attributes.